src/stream/src/executor/managed_state/aggregation/extreme.rs

```diff
@@ -297,15 +310,46 @@ where
             // To future developers: please make **SURE** you have taken `EXTREME_TYPE` into
             // account. EXTREME_MIN and EXTREME_MAX will significantly impact the
             // following logic.
-            let all_data = self.keyspace.scan(self.top_n_count, epoch).await?;
-
-            for (raw_key, mut raw_value) in all_data {
-                // let mut deserializer = value_encoding::Deserializer::new(raw_value);
-                let value = deserialize_cell(&mut raw_value, &self.data_type)
-                    .map_err(StreamExecutorError::serde_error)?;
-                let key = value.clone().map(|x| x.try_into().unwrap());
-                let pks = self.serializer.get_pk(&raw_key[..])?;
-                self.top_n.insert((key, pks), value);
+
+            let all_data_iter = state_table
+                .iter_with_pk_prefix(
+                    self.group_key.as_ref().unwrap_or(&Row::default()),
+                    OrderedRowSerializer::new(vec![
+                        if EXTREME_TYPE == variants::EXTREME_MAX {
+                            OrderType::Descending
+                        } else {
+                            OrderType::Ascending
+                        };
+                        self.group_key
+                            .as_ref()
+                            .map_or(0, |key| key.size())
+                    ]),
+                    epoch,
+                )
+                .await?;
+            pin_mut!(all_data_iter);
+
+            for _ in 0..self.top_n_count.unwrap_or(usize::MAX) {
```

Comment:
If `top_n_count` is `None`, we should keep nothing in the cache.

Reply:
Maybe It should not be a Option... Let's change in other PRs